app/scripts/controllers/buildConfig.js

```diff
@@ -75,18 +75,35 @@ angular.module('openshiftConsole')
     };
 
     $scope.saveEnvVars = function() {
+      $scope.envVarDiffAmount = 0;
       $scope.envVars = _.filter($scope.envVars, 'name');
       buildStrategy($scope.updatedBuildConfig).env = keyValueEditorUtils.compactEntries(angular.copy($scope.envVars));
+
```

Comment:
I like the count, looks helpful.  We have an [environment service](https://github.com/openshift/origin-web-console/blob/master/app/scripts/services/environment.js) now, it might be nice to push the logic up into `EnvironmentService.diff()` or `EnvironmentService.diffCount()`.  

Prob a nit, but I like `_.each()` over `for` loops for readability: 
```JavaScript
_.each(originalEnvs, function(env, i) {
 if(updatedEnvs[i] &&
   env.name !== updatedEnvs[i].name &&
   env.value !== updatedEnvs[i].value) {
   $scope.envVarDiffAmount++;
 }
});     
```
Though `_.reduce` might be a nice way to contain all the pieces:
```JavaScript
      $scope.envVarDiffAmount = _.reduce(originalEnvs, function(count, env, i) {
        if(updatedEnvs[i] &&
          env.name !== updatedEnvs[i].name &&
          env.value !== updatedEnvs[i].value) {
          count++;
        }
        return count;
      }, 0); 
```
We also now have `env.valueFrom` that we are handling in a lot of places.  Might need to add something for that as well.